libr/core/rvc.c

```diff
@@ -730,6 +732,47 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
 		eprintf ("Can't commit");
 		return false;
 	}
+	if (!message) {
+#if __WINDOWS__ //fuck you
+	char *whoami = r_sys_whoami ();
+	if (!whoami) {
+		return false;
+	}
+	char *path = r_str_newf ("c:\\\\users\\%s\\appdata\\local\\tmp\\rvcmessage",
+		whoami);
+	free (whoami);
+#else
+	char *path = r_str_newf ("/tmp/rvcmessage");
+#endif
+	if (!path) {
+		return false;
+	}
+	if (!r_file_touch (path)) {
+		free (path);
+		return false;
+	}
+	if (r_sys_cmdf ("$EDITOR %s", path)) {
+		free (path);
+		return false;
+	}
+	FILE *f;
+	char m[MAX_MESSAGE_LEN + 1];
+	f = fopen (path, "r");
```

Comment:
Use rfileslurp and dont use fixed size buffers

Reply:
There is a reason why I used fixed size buffers and that is because at the end we set message to m and message is constant and can be freed and even if we were to free it we run at the risk of freeing a stack adress at best and a mem leak at worst. I can't find a better solution other than just caping it a 80 chars long which in my opinion at least is better than allowing the user to write an unlimited amount of chars.